ql/pricingengines/asian/turnbullwakemanasianengine.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ Copyright (C) 2019 Quaternion Risk Management Ltd
```

Comment:
I don't see a lot of similar code, though?  Instead of attributing copyright, I would write in the hpp file something like:

"Some parts of the implementation were modeled after calculations from the CommodityAveragePriceOptionAnalyticalEngine class in Open Source Risk Engine (https://github.com/OpenSourceRisk/Engine)."

Reply:
Yes, you're right. Most of the resemblance was lost over time, so I'll update it following your suggestion in the morning. I'll also patch the makefile then. Turns out that fixing things in a hurry makes for poor results...

Reply:
Fixed in 39f16f7